src/pipeImpureRule.ts

```diff
@@ -0,0 +1,61 @@
+import * as Lint from 'tslint/lib/lint';
+import * as ts from 'typescript';
+import {sprintf} from 'sprintf-js';
+import SyntaxKind = require('./util/syntaxKind');
+
+export class Rule extends Lint.Rules.AbstractRule {
+
+    public apply(sourceFile:ts.SourceFile):Lint.RuleFailure[] {
+        return this.applyWithWalker(
+            new ClassMetadataWalker(sourceFile, this));
+    }
+
+    static  FAILURE:string = 'The Pipe decorator of class %s should be pure.';
```

Comment:
That's right, this message is a bit too restrictive :-) I changed it, feel free to modify it to your liking, I'm not good with error messages :-)